### DISCUSSION TITLE
Guidance on Auth digest registration request

### DISCUSSION THREAD

**Discussion by breeblebrax** [sipsorcery-org/sipsorcery#664] [General]
Is there any example out there that shows how do deal with the
case where there is a 
Username = extension
passcode = passcode
Authentication ID = is different than extension?

I have gotten SIPRegistrationUserAgent to work on Asterisk and 3CX.  Calls in both direction, great audio etc.
However 3CX only works if I set the Authentication ID to the Extension(User Name) value.

I did a stepped through the code and found it to be responding with Auth Digest  correctly on the second register attempt, but
it is doesn't know what the Authentication ID is, as I can see where one place it before instantiate SIPRegistrationUserAgent.

Hope that makes sense!

Cheers,
Chuck

**Comment by sipsorcery** [comment]
If you use the [SIPUserAgent.Call overload ](https://github.com/sipsorcery-org/sipsorcery/blob/77894bc3f2025a505039ca446cabf365dd76c74d/src/app/SIPUserAgents/SIPUserAgent.cs#L469) that takes the `SIPCallDescriptor` you can set the [Username and AuthUsername](https://github.com/sipsorcery-org/sipsorcery/blob/77894bc3f2025a505039ca446cabf365dd76c74d/src/app/SIPUserAgents/SIPCallDescriptor.cs#L88) to different values. Sounds like that might resolve your issue.

**Reply by breeblebrax** [reply]
I looked at the member function and figured I either needed to reproduce the code that filled in the other parameters of the SIPCallDescriptor or just expand it with an optional parameter.  If that doesn't fit your approach, let me know and I will do it the other way without the function call change.  Because it is at the end of the function definition and optional with null as a default, it should be backward compatible.  Cheers!

**Reply by breeblebrax** [reply]
Also, it is now working for me for Asterisk and 3CX!  Now, on to RingCentral, which is being very obstinate...
